bolt-modules/ctrl/lib/puppet/functions/ctrl/sleep.rb

```diff
@@ -1,5 +1,15 @@
 # frozen_string_literal: true
 
+def sleep_with_signal_handler(period)
```

Comment:
It feels like this should go into a utility class and be required by both functions, like PuppetX utils in types & providers